chipsec_main.py

```diff
@@ -373,117 +356,49 @@ def run_all_modules(self):
 
         Returns: a pair (continue, exit_code). If continue is False,
           the exit_code should be returned.
         """
-        try:
-            opts, args = getopt.getopt(self.argv, "ip:m:ho:vda:nl:t:j:x:I:",
-            ["ignore_platform", "platform=", "pch=", "module=", "help", "output=",
-              "verbose", "debug", "module_args=", "no_driver", "log=",
-              "moduletype=", "json=", "xml=", "list_tags", "include", "failfast",
-              "no_time", "deltas="])
-        except getopt.GetoptError, err:
-            print str(err)
-            self.usage()
-            return (False, ExitCode.EXCEPTION)
-
-        for o, a in opts:
-            if o in ("-v", "--verbose"):
-                logger().VERBOSE = True
-                logger().HAL     = True
-                logger().DEBUG   = True
-            elif o in ("-d", "--debug"):
-                logger().DEBUG   = True
-            elif o in ("-h", "--help"):
-                self.usage()
-                return (False, ExitCode.OK)
-            elif o in ("-o", "--output"):
-                self._output = a
-            elif o in ("-p", "--platform"):
-                self._platform = a.upper()
-            elif o in ("--pch"):
-                self._pch = a.upper()
-            elif o in ("-m", "--module"):
-                #_module = a.lower()
-                self._module = a
-            elif o in ("-a", "--module_args"):
-                self._module_argv = a.split(',')
-            elif o in ("-i", "--ignore_platform"):
-                logger().log( "[*] Ignoring unsupported platform warning and continue execution" )
-                self._unkownPlatform = False
-            elif o in ("-l", "--log"):
-                #logger().log( "[*] Output to log file '%s' (--log option or chipsec_main.logger().set_log_file in Python console)" % a )
-                logger().set_log_file( a )
-            elif o in ("-t", "--moduletype"):
-                usertags = a.upper().split(",")
-                for tag in usertags:
-                    self.USER_MODULE_TAGS.append(tag)
-            elif o in ("-n", "--no_driver"):
-                self._no_driver = True
-            elif o in ("-x", "--xml"):
-                self._xml_out = a
-            elif o in ("-j", "--json"):
-                self._json_out = a
-            elif o in ("--list_tags"):
-                self._list_tags = True
-            elif o in ("-I","--include"):
-                self.IMPORT_PATHS.append(a)
-            elif o in ("--failfast"):
-                self.failfast = True
-            elif o in ("--no_time"):
-                self.no_time = True
-            elif o in ("--deltas"):
-                self._deltas_file = a
-            else:
-                assert False, "unknown option"
+        parser = argparse.ArgumentParser(prog='chipsec_main.py',usage='%(prog)s [options]',formatter_class=argparse.RawDescriptionHelpFormatter,epilog=ExitCode.Help_String,add_help=False)
```

Comment:
Do we want to leave the .py off the program name.  This may be better for people that install it on a test image.